stdlib/src/collections/list.mojo

```diff
@@ -419,6 +419,35 @@ struct List[T: CollectionElement, hint_trivial_type: Bool = False](
         self.write_to(output)
         return output^
 
+    fn write_bytes(inout self, bytes: Span[Byte, _]):
+        """
+        Write a byte span to this List. `T` **must** be of type `Byte`.
+
+        Args:
+            bytes: The byte span to write to this String. Must NOT be
+              null terminated.
+        """
+        constrained[_type_is_eq[T, Byte]()]()
+        self.extend(rebind[Span[T, bytes.origin]](bytes))
```

Comment:
the extend method has a lot of overhead if you know it's a `DType`. Until we land #3577 this logic will have to stay here
```suggestion
        self.reserve(len(self) + len(other))
        memcpy(
            self.unsafe_ptr() + len(self),
            bytes.unsafe_ptr().bitcast[T](),
            len(bytes),
        )
```